### PR TITLE
Fixing modal example to use hooks

### PR DESCRIPTION
**Pull request by pedroapfilho**
I just updated the modal example to use hooks, nothing much. It's the last example using a class component.

### CODE REVIEW

File: docs/example-react-modal.md [hunk 1]
```diff
@@ -4,63 +4,64 @@ title: Modals
...
+        <hr />
+        <button onClick={onClose}>Close</button>
+      </div>
+    </div>,
+    el
+  )
 }
 
+const rootElement = document.getElementById('root')
+ReactDOM.render(<App />, rootElement)
```
Comment (kentcdodds) [inline docs/example-react-modal.md]: Oh, this shouldn't need to be here.

File: docs/example-react-modal.md [hunk 2]
```diff
@@ -4,63 +4,64 @@ title: Modals
 ---
 
 ```jsx
-import React from 'react'
+import React, { useEffect } from 'react'
 import ReactDOM from 'react-dom'
-import { render, fireEvent } from '@testing-library/react'
```
Comment (kentcdodds) [inline docs/example-react-modal.md]: Why was this removed?

File: docs/example-react-modal.md [hunk 3]
```diff
@@ -4,63 +4,64 @@ title: Modals
...
+      style={{
+        position: 'absolute',
+        top: '0',
+        bottom: '0',
+        left: '0',
+        right: '0',
+        display: 'grid',
+        justifyContent: 'center',
+        alignItems: 'center',
+        backgroundColor: 'rgba(0,0,0,0.3)',
```
Comment (afontcu) [inline docs/example-react-modal.md]: Since we're tweaking this example, would it make sense to remove stylistic CSS to make it more focused on the problem we're trying to solve?
